_includes/locale/Locale.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ */
+
+  declare(strict_types=1);
+
+  namespace Keyman\Site\com\keyman;
+
+  class Locale {
+    public const DEFAULT_LOCALE = 'en';
+
+    public const CROWDIN_LOCALES = array(
+      'en',
+      'es-ES',
+      'fr-FR'
+    );
+
+    // xx-YY locale as specified in crowdin %locale%
+    private static $currentLocale = Locale::DEFAULT_LOCALE;
+
+    // If set to true, warn about untranslated strings
+    private static $debugMode = false; // Keep false in production
+
+    // Associative array to hold translated strings in "current locale" and English
+    private static $langArray = [];
+    private static $langArrayEn = [];
+
+    /**
+     * Return the current locale. Fallback to 'en'
+     * @return $currentLocale
+     */
+    public static function currentLocale() {
+      return Locale::$currentLocale;
+    }
+
+    /**
+     * Validate and override the current locale
+     * @param $locale - the new current locale (xx-YY as specified in crowdin %locale%)
+     */
+    public static function overrideCurrentLocale($locale) {
+      if (Locale::validateLocale($locale)) {
+        Locale::$currentLocale = $locale;
+      }
+    }
+
+    /**
+     * Validate $locale is an acceptable locale.
+     * Using xx-YY as specified in crowdin %locale%
+     * @param $locale - the locale to validate
+     * @return true if valid locale
+     */
+    public static function validateLocale($locale) {
+      return in_array($locale, Locale::CROWDIN_LOCALES);
+    }
```

Comment:
This may not be viable because we may not have localizations for a given l10n domain.

Reply:
Can you give an example? 
We can set the list of available languages on
https://crowdin.com/project/keymancom/settings#languages

(would need the sil_ltops login to add custom languages)

Reply:
removed with the glob refactor.